internal/csi/csi.go

```diff
@@ -0,0 +1,174 @@
+package csi
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-k8s-tester/ec2config"
+	"github.com/aws/aws-k8s-tester/internal/ec2"
+	"github.com/aws/aws-k8s-tester/internal/ssh"
+
+	"go.uber.org/zap"
+)
+
+type Tester struct {
+	cfg *ec2config.Config
+	iam *iamResources
+	ec  ec2.Deployer
+
+	terminateOnExit bool
+	journalctlLogs  bool
+}
+
+func NewTester(cfg *ec2config.Config, terminateOnExit, journalctlLogs bool) (tester *Tester, err error) {
+	// TODO: some sort of validations check
+	tester = &Tester{
+		terminateOnExit: terminateOnExit,
+		journalctlLogs:  journalctlLogs,
+		cfg:             cfg,
+	}
+
+	permissions, permissionsErr := createPermissions(tester.cfg)
```

Comment:
Move line 33-36 to line 48 (after `ec.Create`), so that we can pass logger to `createPermissions`

```go
permissions, permissionsErr := createPermissions(tester.ec.Logger(), tester.cfg)
```

We should

```go
func createPermissions(lg *zap.Logger, cfg *ec2config.Config) (resources *iamResources, err error) {
	resources, err = createIAMResources(lg, cfg.AWSRegion)

func createIAMResources(lg *zap.Logger, awsRegion string) (resources *iamResources, err error) {
```

And 

```diff
-	// Creates new logger
-	lg, err := zap.NewProduction()
-	if err != nil {
-		return nil, fmt.Errorf("failed to create logger (%v)", err)
-	}
```

in line 106 `iam.go` https://github.com/aws/aws-k8s-tester/pull/13/files#diff-8e258d08d765f8b8e6d1b80300b2f749R106

Reply:
We decided not to do this for now due to the need of permissions for NewDeployer() and Create() but may explore this in the future after some improvements to the implementation of tester.